### PR TITLE
Sound volume adjustment

### PR DESCRIPTION
**Pull request by xieWright**
Added a volume bar that adjusts the volume of sound in the game. Added Doxygen comments and tests accordingly.

### CODE REVIEW

File: assets/volume_slider.png
Comment (Lurk390) [inline assets/volume_slider.png:1]: I'm not really a fan of this graphic, I think a simple horizontal line would suffice and is intuitive.

File: tempus_fugit_minecraft/sound.py [hunk 1]
```diff
@@ -1,37 +1,35 @@
...
-    @return rock_hit_sound An instance of Sound class.
-    """
     def __init__(self, file_path=None):
-        """!
-        @brief Initializes the Sound class.
-        @details If a file path is named, uses that to load the sound file        
-        @param file_path string The file path for the sound file
-        @see [Issue#17](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/17)
-        """
+        '''!Initializes the class Sound. If a file path is named, uses that to load the sound file
```
Comment (Lurk390) [inline tempus_fugit_minecraft/sound.py:7]: Keep the `"""!` on its own line for readability

File: tempus_fugit_minecraft/sound.py [hunk 2]
```diff
@@ -1,37 +1,35 @@
...
         if file_path != None:
             self.sound_file = media.load(file_path)
 
     def load_sound(self, file_path):
-        """!
-        @brief Loads the sound into the media player.
-        @param file_path string The file path for the sound file
-        @see [Issue#17](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/17)
-        """
+        '''!Loads the sound into the media player
```
Comment (Lurk390) [inline tempus_fugit_minecraft/sound.py:22]: Keep the """! on its own line for readability

File: tempus_fugit_minecraft/sound.py [hunk 3]
```diff
@@ -1,37 +1,35 @@
...
+            @see [issue#17] (https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/17)
+        '''
         self.sound_file = media.load(file_path)
 
     def play_sound(self):
-        """!
-        @brief plays the sound contained in the class.
-        @see [Issue#17](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/17)
-        """
+        '''!Plays the sound file contained in the class
```
Comment (Lurk390) [inline tempus_fugit_minecraft/sound.py:29]: Keep the """! on its own line for readability

File: tempus_fugit_minecraft/sound.py [hunk 4]
```diff
@@ -42,11 +40,14 @@ def play_sound(self):
...
-    """!
-    @brief Subclass of sound for sounds that will be repeated
-    @return wind_blowing An instance of BackgroundSound class.
-    """
+    '''Subclass of sound for sounds that will loop'''
     def __init__(self, file_path=None):
         super().__init__(file_path)
-        self.player.loop = True
\ No newline at end of file
+        '''!Initializes the class BackgroundSound. This is to separate sounds that will loop and have different
```
Comment (Lurk390) [inline tempus_fugit_minecraft/sound.py:46]: Keep the """! on its own line for readability

File: tempus_fugit_minecraft/sound_list.py [hunk 5]
```diff
@@ -1,10 +1,84 @@
 from tempus_fugit_minecraft import sound
 """!
-@brief File to contain all sound variables
+@brief File to contain all sound variables and the SoundList class that will contain and adjust volume controls.
 """
```
Comment (Lurk390) [inline tempus_fugit_minecraft/sound_list.py:4]: This should be at the top of the file above the imports

File: tempus_fugit_minecraft/sound_list.py [hunk 6]
```diff
@@ -1,10 +1,84 @@
...
 
+class SoundList():
```
Comment (Lurk390) [inline tempus_fugit_minecraft/sound_list.py:6]: Need a class docstring here

File: tempus_fugit_minecraft/window.py [hunk 7]
```diff
@@ -60,6 +63,17 @@ def __init__(self, *args, **kwargs):
 
         self.paused = False
 
+        """Solves issue #"""
```
Comment (Lurk390) [inline tempus_fugit_minecraft/window.py:66]: This comment looks incomplete, I'd either get rid of it or if you finish it use a normal comment with a `#`

**Review by Lurk390 — CHANGES_REQUESTED** [review]
Main things is I don't really like how the slider looks (straight line would be cleaner imo) and some documentation changes.

File: assets/volume_slider.png
Comment (xieWright) [reply assets/volume_slider.png:1]: It probably would work too. Honestly speaking, I'm not much of an artist and trying to do the actual painting in pyglet is rough. The image itself is pretty replaceable.

File: tempus_fugit_minecraft/shaders.py [hunk 8]
```diff
@@ -97,6 +97,7 @@ def adjust_ambient_light(self, red, green, blue):
         @Param blue a number that specifies the intensity of blue light
         @return returns the c_float vector that corresponds to the ambient
         @see [Issue#12](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/12)
+        @see [Issue#18](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/18)
```
Comment (w1158mxs) [inline tempus_fugit_minecraft/shaders.py:100]: issue #18 contain 
Creation of Sun object
Creation of Moon object
I suggest editing that while will not do it

File: tempus_fugit_minecraft/window.py [hunk 9]
```diff
@@ -57,8 +60,29 @@ def __init__(self, *args, **kwargs):
...
-
+        
+        """Solves issue #99. Properties that are related to volume adjustments"""
+        self.volume_slider_image = load('assets/volume_slider.png')
+        self.volume_knob_image = load('assets/volume_knob.png')
+        self.volume_control_batch = pyglet.graphics.Batch()
+        self.volume_control_back = pyglet.graphics.OrderedGroup(1)
+        self.volume_control_front = pyglet.graphics.OrderedGroup(0)
+        self.volume_slider_sprite = Sprite(self.volume_slider_image, 
+                                           x=WINDOW_WIDTH // 16,
```
Comment (w1158mxs) [inline tempus_fugit_minecraft/window.py:71]: Can you keep space between (x and =)
 x = WINDOW_WIDTH

Comment (w1158mxs) [reply tempus_fugit_minecraft/window.py:71]: same for y

**Review by w1158mxs — APPROVED** [review]
Small change - looks good! Thank you

**Review by bakursait2 — APPROVED** [review]
All looks good to me.

**Review by criticalglitch — APPROVED** [review]

**Pull request merged**